CMakeLists.txt

```diff
@@ -1,5 +1,14 @@
 cmake_minimum_required(VERSION 3.2)
 
+# Compile with /usr/bin/clang on MacOS
+# See https://github.com/diffblue/cbmc/issues/4956
+if (CMAKE_HOST_SYSTEM_NAME STREQUAL Darwin)
+    if(NOT DEFINED CMAKE_C_COMPILER)
+        message(STATUS "Setting CMAKE_C_COMPILER to /usr/bin/clang for MacOS")
+        set(CMAKE_C_COMPILER "/usr/bin/clang")
```

Comment:
This is a nice change - it's already documented in our [`COMPILING.md`](https://github.com/diffblue/cbmc/blob/develop/COMPILING.md) file that we need to do this by hand on MacOS, so I'm all for the automation here 👍🏻